base_flexcons.py

```diff
@@ -0,0 +1,325 @@
+from abc import abstractmethod
+from typing import Dict, List, Optional
+
+import numpy as np
+from selfTrainingInitial import SelfTrainingClassifier
+from sklearn.base import clone
+from sklearn.metrics import accuracy_score
+from sklearn.utils import safe_mask
+
+
+class BaseFlexCon(SelfTrainingClassifier):
+    """
+    Classe que serve como base para os diferentes métodos FlexCon
+
+    Args:
+        SelfTrainingClassifier: Método SelfTraining
+    """
+    def __init__(
+        self,
+        base_classifier,
+        threshold=0.95,
+        verbose=False,
+        max_iter=10
+    ):
+        self.base_classifier = base_classifier
+        self.threshold = threshold
+        self.verbose = verbose
+        self.max_iter = max_iter
+        self.pred_x_it = {}
+        self.cl_memory = []
+        self.classifier_ = clone(base_classifier)
+        self.accuracy_history = []
```

Comment:
Seria muito bom a gente indicar qual seria o tipo que a gente está tratando. Por exemplo:
```suggestion
        threshold: float = 0.95,
        verbose: bool = False,
        max_iter: int = 10
    ):
        self.base_classifier = base_classifier
        self.threshold = threshold
        self.verbose = verbose
        self.max_iter = max_iter
        self.pred_x_it: dict[int, int] = {}
        self.cl_memory = []
        self.classifier_ = clone(base_classifier)
        self.accuracy_history: list[float] = []
```

continue a partir daí, coloque, de preferência o tipo correto para os argumentos que são listas, dicionários e etc. VALE PARA TODO O RESTO DO ARQUIVO!

Reply:
Irei repassar por todo o arquivo.